src/selectors/user-list/index.js

```diff
@@ -0,0 +1,24 @@
+export const getFetching = (state) => {
+    let { fetching } = selectUserFromState(state);
+    return fetching;
+}
+export const getSearch = (state) => {
+    let { search } = selectUserFromState(state);
+    return search;
+}
+export const getUsers = (state) => {
+    let { users, search, sort } = selectUserFromState(state);
+    let tUsers = (search) ? getFilterUser(users, search) : state.user.users;
+    return (sort) ? getSortUser(tUsers) : tUsers;
+}
+export const getFilterUser = (users, search) => {
+    let tUsers = [...users];
+    return tUsers.filter(u => ((u.login.indexOf(search) > -1) && u));
```

Comment:
`map`, `reduce`, `filter` functions return a new array as an output. This means you can apply these functions directly to the source object. so the function would be:
```
export const getFilterUser = (users, search) => users.filter(u => ((u.login.indexOf(search) > -1) && u));
```